### ISSUE TITLE
Embedding was useful, is there an alternative?

### ISSUE THREAD

**Issue by rlaphoenix** [construct/construct#1027]
For example, a Sample Entry MP4 Box could be defined like so in 2.9.x,

```py
SampleEntryBox = Prefixed(Int32ub, Struct(
    "format" / PaddedString(4, "ascii"),
    Padding(6, pattern=b"\x00"),
    "data_reference_index" / Default(Int16ub, 1),
    Embedded(Switch(this.format, {
        "ec-3": MP4ASampleEntryBox,
        "mp4a": MP4ASampleEntryBox,
        "enca": MP4ASampleEntryBox,
        "avc1": AVC1SampleEntryBox,
        "encv": AVC1SampleEntryBox
    }, Struct("data" / GreedyBytes)))
), includelength=True)
```

Now, to get the same result, you need to use If statements for each entry, with a very repetitive final if statement for that fallback of data/GreedyBytes:

```py
SampleEntryBox = Prefixed(Int32ub, Struct(
    "format" / PaddedString(4, "ascii"),
    Padding(6, pattern=b"\x00"),
    "data_reference_index" / Default(Int16ub, 1),
    "ec-3" / If(this.format == "ec-3", MP4ASampleEntryBox),
    "mp4a" / If(this.format == "mp4a", MP4ASampleEntryBox),
    "enca" / If(this.format == "enca", MP4ASampleEntryBox),
    "avc1" / If(this.format == "avc1", AVC1SampleEntryBox),
    "encv" / If(this.format == "encv", AVC1SampleEntryBox),
    "data" / If(this.format not in ("ec-3", "mp4a", "enca", "avc1", "encv"), GreedyBytes)
), includelength=True)
```

I understand you stated many times here that you removed Embedding as the code for it was hacky. However, it was an extremely useful feature however hacky it was and I cannot find an alternative to use when upgrading pymp4 from 2.8 to 2.10. Some alternative method to Embedding would be appreciated.

**Comment by arekbulski** [comment]
No, there is no alternative. The recommended way of going about it is nesting and not embedding sub-structs. If you want embedding then fork the project and deal with this shit, I did once and did not like it. Embedding half works, half the time, and half the right way.

**Comment by rvanlaar** [comment]
@rlaphoenix 

I'm looking into parsing QuickTime files and noticed the same challenge with construct 2.10.
Did you end up finding a solution?

**Comment by rlaphoenix** [comment]
I have not found a solution that works quite the same way, no.

**Comment by arekbulski** [comment]
TLDR: No, there is no substitute for Embedded(anything).
